### PR TITLE
reduce amount of data returned by ui5.control.getAggregationProperty

### PR DESCRIPTION
**Pull request by heraldantony**
The data returned after decycling cyclic references is too big, and the call fails.
Remove unnecessary properties "oParent", "oPropagatedProperties" and "undefined" from returned data

### CODE REVIEW

File: src/reuse/modules/ui5/types/ui5.types.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export type Ui5Selector = {
+  elementProperties: any
+};
+export type Ui5SelectorOptions = {
```
Comment (Templada) [inline src/reuse/modules/ui5/types/ui5.types.ts:4]: @DIFSRIP should we rename SelectorOptions?

File: src/scripts/hooks/utils/cycle.js [hunk 2]
```diff
@@ -104,10 +102,16 @@ module.exports = {
 
           nu = {};
```
Comment (Templada) [inline src/scripts/hooks/utils/cycle.js:103]: what stands 'nu' for?

File: src/scripts/hooks/utils/locatorCommands.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ module.exports = {
     return lib.uiControlExecuteLocator(ui5Selector, null, timeout, rootElement);
   },
 
-  getUI5Property: async function (propName, selectorOrElement) {
+  getUI5Property: async function (propName, element) {
```
Comment (Templada) [inline src/scripts/hooks/utils/locatorCommands.js:14]: why did we change to element only here? Is it just bc we already got the element at this level?

**Review by DIFSRIP — COMMENTED** [review]
please see my comments

File: src/reuse/modules/ui5/types/ui5.types.ts (hunk 1 above)
Comment (Templada) [reply src/reuse/modules/ui5/types/ui5.types.ts:4]: agree. @heraldantony Can you change to "Ui5SelectorWithOptions" please?

File: src/reuse/modules/ui5/types/ui5.types.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export type Ui5Selector = {
+  elementProperties: any
```
[the remark this answers is not in the document]
Comment (Templada) [reply src/reuse/modules/ui5/types/ui5.types.ts:2]: +1

**Review by Templada — CHANGES_REQUESTED** [review]
@heraldantony please see the comments. A few naming changes are requested.

File: src/scripts/hooks/utils/locatorCommands.js (hunk 3 above)
Comment (heraldantony) [reply src/scripts/hooks/utils/locatorCommands.js:14]: will fix that, that code shouldn't have been changed

File: src/reuse/modules/ui5/types/ui5.types.ts (hunk 1 above)
Comment (heraldantony) [reply src/reuse/modules/ui5/types/ui5.types.ts]: yes, changed to Ui5SelectorWithOptions

File: src/scripts/hooks/utils/cycle.js (hunk 2 above)
Comment (heraldantony) [reply src/scripts/hooks/utils/cycle.js:103]: don't know why :)  Code comes from here - https://github.com/douglascrockford/JSON-js/blob/master/cycle.js, and this is also referred to in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cyclic_object_value

File: src/reuse/modules/ui5/types/ui5.types.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (heraldantony) [reply src/reuse/modules/ui5/types/ui5.types.ts]: Added optional values, and fixed types.

**Review by DIFSRIP — APPROVED** [review]
lgtm

**Review by Templada — APPROVED** [review]

**Pull request merged**